pkg/pgp/pgp.go

```diff
@@ -138,9 +103,40 @@ func DecodePgpKey(pgpKey, path string) (*openpgp.Entity, error) {
 
 // DecodeAndArmorBase64Entity decodes a base64 encoded entity and armors it.
 func DecodeAndArmorBase64Entity(encodedEntity string, armorType string) (string, error) {
-	decodedEntity, err := base64.StdEncoding.DecodeString(encodedEntity)
+	decodedEntity, err := decodeAndValidatePGPkey(encodedEntity)
 	if err != nil {
 		return "", err
 	}
 	return parmor.ArmorWithType([]byte(decodedEntity), armorType)
 }
+
+// decodeAndValidatePGPkey PGP key is a wrapper function
+// around golang standard base64 package
+// which will validate keys with checksum as well as non checksum and return decoded bytes from key part
+func decodeAndValidatePGPkey(encodedKeyData string) ([]byte, error) {
```

Comment:
where is the validate part in here? I think if line 118 succeed's its decoding only?